packages/cli/test/integration/generators/controller.integration.js

```diff
@@ -1,425 +1,313 @@
-      )
-      .then(() => {
-        assert.noFile(withInputName);
-      });
+    return expect(
+      testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () =>
+          testUtils.givenLBProject(SANDBOX_PATH, {excludeKeyword: true}),
+        )
+        .withPrompts(basicCLIInput),
+    ).to.be.rejectedWith(/No `loopback` keyword found in/);
   });
 
-  describe('basic', () => {
-    describe('with input', () => {
-      let tmpDir;
-      before(() => {
-        return testUtils
-          .runGeneratorWith(generator, withInputProps, dir => {
-            tmpDir = dir;
-            testUtils.givenAnApplicationDir(dir);
-          })
-          .toPromise();
-      });
+  describe('basic controller', () => {
+    it('scaffolds correct file with input', async () => {
+      await testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () => testUtils.givenLBProject(SANDBOX_PATH))
+        .withPrompts(basicCLIInput);
 
-      it('writes correct file name', () => {
-        assert.file(tmpDir + withInputName);
-        assert.noFile(tmpDir + templateName);
-      });
-
-      it('scaffolds correct files', () => {
-        checkBasicContents(tmpDir);
-      });
+      assert.file(expectedFile);
+      checkBasicContents();
     });
-    describe('with arg', () => {
-      let tmpDir;
-      before(() => {
-        return testUtils
-          .runGeneratorWith(generator, null, dir => {
-            tmpDir = dir;
-            testUtils.givenAnApplicationDir(dir);
-          })
-          .withArguments('productReview');
-      });
 
-      it('writes correct file name', () => {
-        assert.file(tmpDir + withInputName);
-        assert.noFile(tmpDir + templateName);
-      });
+    it('scaffolds correct file with args', async () => {
+      await testUtils
+        .executeGenerator(generator)
+        .inDir(SANDBOX_PATH, () => testUtils.givenLBProject(SANDBOX_PATH))
+        .withArguments('productReview');
 
-      it('scaffolds correct files', () => {
-        checkBasicContents(tmpDir);
-      });
+      assert.file(expectedFile);
+      checkBasicContents();
     });
   });
 
-  describe('REST CRUD', () => {
-    const baseInput = {
-      name: 'productReview',
-      controllerType: ControllerGenerator.REST,
-    };
-
-    it('creates REST CRUD template with valid input', () => {
-      let tmpDir;
-      return testUtils
-        .runGeneratorWith(
-          generator,
-          Object.assign(
-            {
-              modelName: 'ProductReview',
-              repositoryName: 'BarRepository',
-              id: 'number',
-            },
-            baseInput,
-          ),
-          dir => {
-            tmpDir = dir;
-            givenModelAndRepository(tmpDir);
-          },
+  describe('REST CRUD controller', () => {
+    const restCLIInputComplete = Object.assign(
+      {},
+      {
+        modelName: 'ProductReview',
+        repositoryName: 'BarRepository',
```

Comment:
I think this might be missing key/value pair `{id: 'number'}`, but if the tests pass, maybe we don't need it?

Reply:
`restCLIInput` ... the object this is merged with using `Object.assign` has `id: 'number'`.